mobly/controllers/android_device.py

```diff
@@ -812,12 +826,16 @@ def reboot(self):
         if self.is_bootloader:
             self.fastboot.reboot()
             return
+        snippet_info = self._get_active_snippet_info()
         self.stop_services()
         self.adb.reboot()
         self.wait_for_boot_completion()
         if self.is_rootable:
             self.root_adb()
-        self.start_services(skip_sl4a=getattr(self, KEY_SKIP_SL4A, False))
+        skip_sl4a = getattr(self, KEY_SKIP_SL4A, False) or self.sl4a is None
```

Comment:
sl4a is required by default is an incorrect assumption as our usage is expanded beyond Nexus devices. In the open source world or NBU use cases, sl4a is not there by default. Also in console mode, sl4a is not started by default either. So overall this is a better check than the original one. The old logic causes errors in reboot in python console if sl4a is not installed on the device at all.

We might want to consider making sl4a a service that needs to be explicitly requested like a snippet as opposed to assuming it's there by default.